### PR TITLE
[empty-dir] Adds support for arrays

### PR DESCRIPTION
**Pull request by joshuaharry**
Please fill in this template.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: See https://github.com/gulpjs/empty-dir/blob/4817e656907fe58fcb3e4b41fde7363812a8f737/test/index.js#L36

### CODE REVIEW

File: types/empty-dir/empty-dir-tests.ts [hunk 1]
```diff
@@ -7,6 +7,9 @@ emptyDir('./', (err, isEmpty) => {
     isEmpty;
 });
 
+// $ExpectType boolean
+emptyDir.sync(["./test", "./array"]);
```
Comment (BendingBender) [inline types/empty-dir/empty-dir-tests.ts:11]: Please also add an analogous test for the non-sync version.

File: types/empty-dir/index.d.ts [hunk 2]
```diff
@@ -9,19 +9,19 @@
 export = emptyDir;
 
 declare function emptyDir(
-    dir: string,
+    dir: string | string[],
```
Comment (BendingBender) [inline types/empty-dir/index.d.ts:12]: Please use `readonly` arrays for inputs that are not modified by a function:
```suggestion
    dir: string | readonly string[],
```

**Review by BendingBender — CHANGES_REQUESTED** [review]

File: types/empty-dir/empty-dir-tests.ts (hunk 1 above)
Comment (joshuaharry) [reply types/empty-dir/empty-dir-tests.ts:11]: Added on line 13

File: types/empty-dir/empty-dir-tests.ts [hunk 3]
```diff
@@ -7,6 +7,14 @@ emptyDir('./', (err, isEmpty) => {
...
+
+emptyDir(['x'], (err: Error, isEmpty: boolean) => {
+  if (err) return err;
+  return isEmpty;
```
Comment (BendingBender) [inline types/empty-dir/empty-dir-tests.ts:15]:
```suggestion
emptyDir(['x'], (err, isEmpty) => {
    // $ExpectType ErrnoException
    err;
    // $ExpectType boolean
    isEmpty;
```

File: types/empty-dir/index.d.ts [hunk 4]
```diff
@@ -9,19 +9,19 @@
...
 ): void;
 declare function emptyDir(
-    dir: string,
+    dir: string | readonly string[],
     filter?: (path: string) => boolean
 ): Promise<boolean>;
 
 declare namespace emptyDir {
-    function sync(dir: string, filter?: (path: string) => boolean): boolean;
+    function sync(dir: string | string[], filter?: (path: string) => boolean): boolean;
```
Comment (BendingBender) [inline types/empty-dir/index.d.ts:26]:
```suggestion
    function sync(dir: string | readonly string[], filter?: (path: string) => boolean): boolean;
```

**Review by BendingBender — CHANGES_REQUESTED** [review]

**Review by BendingBender — APPROVED** [review]

**Review by djcsdy — APPROVED** [review]

**Comment by djcsdy** [comment]
Ready to merge

**Pull request merged**